### PR TITLE
exclude tags and reservedConcurrency from version hash

### PR DESCRIPTION
**Pull request by thewizarodofoz**
Closes: #7822

### CODE REVIEW

File: lib/plugins/aws/package/compile/functions/index.js [hunk 1]
```diff
@@ -494,7 +494,12 @@ class AwsCompileFunctions {
             });
         }).then(() => {
           // Include function configuration in version id hash (without the Code part)
-          const properties = _.omit(_.get(functionResource, 'Properties', {}), 'Code');
+          const functionWideProperties = ['ReservedConcurrentExecutions', 'Tags'];
+          const properties = _.omit(
+            _.get(functionResource, 'Properties', {}),
+            'Code',
+            ...functionWideProperties
+          );
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/functions/index.js:502]: What's the added value of creating `functionWideProperties` variable.

Isn't it simpler to just do `_.omit(..., ['Code', 'ReservedConcurrentExecutions', 'Tags']` ?

File: lib/plugins/aws/package/compile/functions/index.test.js [hunk 2]
```diff
@@ -2240,6 +2240,55 @@ describe('AwsCompileFunctions', () => {
         });
     });
 
+    it('should not create a new version object if only function-wide configuration changed', () => {
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/functions/index.test.js:2243]: All new tests should be written with help of `runServerless` util (see https://github.com/serverless/serverless/blob/master/test/README.md)

In this case I would rely on basic `function` fixture as follows

1. Setup a fixture with `fixture.setup`, as e.g. it's done here: https://github.com/serverless/serverless/blob/d9b91e97fb81b6f19c9f95920b509d623bdca37d/lib/utils/analytics/generatePayload.test.js#L18
2. Issue first `runServerless` and read all generated lambda version ids
3. Modify `reservedConcurrency` and `tags` with help of `updateConfig` (returned by `fixture.setup()`) as it's done here: https://github.com/serverless/serverless/blob/d9b91e97fb81b6f19c9f95920b509d623bdca37d/test/integration/apiGateway.test.js#L191-L204
4. Re-run `runServerless`, and confirm that version id didn't change

**Review by medikoo — CHANGES_REQUESTED** [review]
Thank you @thewizarodofoz ! That looks very promising. Please see my remarks

File: lib/plugins/aws/package/compile/functions/index.js (hunk 1 above)
Comment (thewizarodofoz) [reply lib/plugins/aws/package/compile/functions/index.js:502]: I thought it would make it more clear as to why these fields are omitted

Comment (medikoo) [reply lib/plugins/aws/package/compile/functions/index.js:502]: @thewizarodofoz that's a good point , but it'll probably be fine to solve it with simple inline code comment:

```javascript
_.omit(..., [
  'Code',
   // Properties applied to function globally (not specific to version or alias)
  'ReservedConcurrentExecutions', 'Tags'
])
```

File: lib/plugins/aws/package/compile/functions/index.test.js (hunk 2 above)
Comment (thewizarodofoz) [reply lib/plugins/aws/package/compile/functions/index.test.js]: Thanks @medikoo [I refactored the test](https://github.com/serverless/serverless/pull/8212/commits/8562f4f598675a2791fd444e5db8dbbdbd680227), but it seems to me the second call to `runServerless` isn't reading the updated config. I debugged the test and I did see `updateConfig` updating the `serverless.yml` in the temp dir, but as you can see in [this assertion (which fails)](https://github.com/thewizarodofoz/serverless/commit/8562f4f598675a2791fd444e5db8dbbdbd680227#diff-587fde9214d0bca41ecd7959b77058d6R2942), `runServerless` isn't picking up those changes.

**Review by medikoo — COMMENTED** [review]
> `runServerless` isn't picking up those changes.

Thanks for pointing. Indeed there was an issue in memoization setup, due to which, in same process second `serverless` instance of same service was relying on previously resolved config. I've fixed that in https://github.com/serverless/serverless/pull/8231

Update with master and issue should be gone

File: lib/plugins/aws/package/compile/functions/index.test.js [hunk 3]
```diff
@@ -2831,4 +2832,40 @@ describe('AwsCompileFunctions #2', () => {
       });
     });
   });
+
+  describe('function config', () => {
+    it('should not create a new version object if only function-wide configuration changed', () => {
+      return fixtures.setup('function').then(({ servicePath, updateConfig }) =>
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/functions/index.test.js:2838]: Can we upgrade this to async/await syntax?

**Review by medikoo — COMMENTED** [review]
@thewizarodofoz that looks great! Thank you

I have just one final remark. Last week we've released v2 and dropped support for Node.js v6, which means we can freely now use async/await syntax, and ideally (for better maintenance) if all new code is introduced in that style.

Can you update the fixture so it's configured with _async_ function and `await` keywords?

Sorry for late notice

File: lib/plugins/aws/package/compile/functions/index.test.js (hunk 3 above)
Comment (thewizarodofoz) [reply lib/plugins/aws/package/compile/functions/index.test.js:2838]: I was really hoping to do that! Yay!

**Comment by medikoo** [comment]
@thewizarodofoz is this ready for re-review? If so, please re-request review in _Reviewers_ box

File: lib/plugins/aws/package/compile/functions/index.test.js [hunk 4]
```diff
@@ -2831,4 +2832,38 @@ describe('AwsCompileFunctions #2', () => {
       });
     });
   });
+
+  describe.only('function config', () => {
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/functions/index.test.js:2836]: `.only` I believe was committed in  by accident

**Review by medikoo — CHANGES_REQUESTED** [review]
Thank you @thewizarodofoz, It's all great, aside of that we should not commit `describe.only`.

Can you update it?

**Review by medikoo — APPROVED** [review]
Thank you @thewizarodofoz !

**Comment by azizur** [comment]
This PR has not resolved the issue. I am still getting `A version for this Lambda function exists ( XXX ). Modify the function to create a new version.`.

**Comment by medikoo** [comment]
@azizur see: https://github.com/serverless/serverless/issues/7822#issuecomment-699940541

**Pull request merged**